File: Wdtc/src/main/java/org/wdt/wdtc/ui/window/ExceptionWindow.java
package org.wdt.wdtc.ui.window;

import javafx.application.Platform;
import javafx.scene.Scene;
import javafx.scene.control.TextArea;
import javafx.scene.image.Image;
import javafx.scene.layout.VBox;
import javafx.scene.text.Font;
import javafx.stage.Stage;
import org.apache.log4j.Logger;
import org.wdt.wdtc.core.utils.WdtcLogger;

public class ExceptionWindow {
  private static final Logger logmaker = WdtcLogger.getLogger(ExceptionWindow.class);


  public static void setErrorWin(Throwable e) {
    logmaker.error("Error", e);
    setWin(WdtcLogger.getExceptionMessage(e), "发生错误!");
  }

  public static void setWin(String e, String title) {
    Runnable runnable = (() -> {
      Stage stage = new Stage();
      stage.setWidth(1000);
      stage.setHeight(600);
      TextArea label = new TextArea();
      label.setText(e);
      label.setFont(new Font(14));
      label.setPrefHeight(stage.getHeight());
      label.setPrefWidth(stage.getWidth());
      VBox vBox = new VBox();
      stage.setTitle(title);
      stage.getIcons().add(new Image("assets/icon/ico.jpg"));
      WindwosSizeManger size = new WindwosSizeManger(stage);
      size.ModifyWindwosSize(vBox, label);
      stage.setScene(new Scene(vBox));
      stage.show();
      stage.setOnCloseRequest(windowEvent -> vBox.getChildren().clear());
    });
    Platform.runLater(runnable);
  }
}

File: Wdtc/src/main/kotlin/org/wdt/wdtc/ui/window/ExceptionWindow.kt
package org.wdt.wdtc.ui.window

import org.wdt.wdtc.core.utils.error
import org.wdt.wdtc.core.utils.getExceptionMessage
import org.wdt.wdtc.core.utils.logmaker
import java.awt.BorderLayout
import java.awt.Color
import java.awt.Font
import javax.swing.JFrame
import javax.swing.JScrollPane
import javax.swing.JTextArea

fun setErrorWin(e: Throwable) {
  e.let {
    logmaker.error(it)
    setWin(it.getExceptionMessage(), "发生错误!")
  }
}

fun setWin(message: String, title: String) {
  val textArea = JTextArea().apply {
    text = message
    setCaretPosition(0)
    setCaretColor(Color.RED)
    setBackground(Color(0xEEEEEE))
    setFont(Font("JetBrains Mono", Font.ITALIC, 14))
    setSelectionColor(Color.BLACK)
    setSelectedTextColor(Color.WHITE)
    setLineWrap(true)
    setWrapStyleWord(true)
  }
  JFrame(title).apply {
    layout = BorderLayout()
    add(JScrollPane(textArea), BorderLayout.CENTER)
    setTitle(title)
    isVisible = true
    defaultCloseOperation = JFrame.EXIT_ON_CLOSE
    setSize(1000, 500)
  }.run {
    setResizable(true)
  }

}
